config/stored_requests.go

```diff
@@ -57,10 +57,10 @@ type StoredRequests struct {
 	// Files should be used if Stored Requests should be loaded from the filesystem.
 	// Fetchers are in stored_requests/backends/file_system/fetcher.go
 	Files FileFetcherConfig `mapstructure:"filesystem"`
-	// Postgres configures Fetchers and EventProducers which read from a Postgres DB.
-	// Fetchers are in stored_requests/backends/db_fetcher/postgres.go
-	// EventProducers are in stored_requests/events/postgres
-	Postgres PostgresConfig `mapstructure:"postgres"`
+	// Database configures Fetchers and EventProducers which read from a Database.
+	// Fetchers are in stored_requests/backends/db_fetcher/fetcher.go
+	// EventProducers are in stored_requests/events/database
+	Database DatabaseConfig `mapstructure:"database"`
```

Comment:
This change looks good, however I think we need to be backwards compatible. Hosts with stored requests housed in a database will have the `postgres` configuration so this would be a breaking change as currently written.

I'd like to see an approach here that we've used quite a bit where we map the old field to the new field if the old field is present and the new field isn't defined. If both the old and new field are defined, we ignore the old field but issue a warning.

Take a look at `config/config.go#migrateConfigPurposeOneTreatment` or `config/config.go#migrateConfigSpecialFeature1` for reference.

This approach will allow us to give hosts some time to make the necessary adjustments to their config. We will eventually remove the backwards compatibility logic.

Reply:
Good idea. I will follow the proposed approach.